firebase-perf/e2e-app/src/androidTest/java/com/google/firebase/testing/fireperf/FirebasePerformanceFragmentScreenTracesTest.java

```diff
@@ -54,10 +54,11 @@ public void scrollAndCycleThroughAllFragments() throws InterruptedException {
     scenario.onActivity(new NavigateAction(R.id.navigation_slow));
     scrollRecyclerViewToEnd(SlowFragment.NUM_LIST_ITEMS, R.id.rv_numbers_slow);
     assertThat(scenario.getState()).isEqualTo(State.RESUMED);
-    scenario.moveToState(State.STARTED).moveToState(State.CREATED);
-
-    // End Activity screen trace by relaunching the activity to ensure the screen trace is sent.
-    scenario.launch(FirebasePerfFragmentsActivity.class);
+    scenario.moveToState(State.STARTED).moveToState(State.CREATED); // trigger activity screen trace
+    // Wait for TransportManager and Firelog executors to finish
+    Thread.sleep(5000);
+    // Block until all Fireperf events are sent by Firelog
+    InstrumentationTestUtil.flgForceUploadSync();
```

Comment:
Would the screen trace be completed by this time? Should we move to a different activity to make sure the current activity trace is completed?

Reply:
The line before this is
```scenario.moveToState(State.STARTED).moveToState(State.CREATED); // trigger activity screen trace```
which already triggers screen trace to be recorded. Reading the source code, behind the scenes that method also moves to different activity.

Reply:
Sounds good! The activity transition was not evidently understandable in the code here. May be a comment here would make it easy for understanding this for future contributions.